src/Selenium2Library/keywords/_element.py

```diff
@@ -479,17 +479,27 @@ def simulate(self, locator, event):
     def press_key(self, locator, key):
         """Simulates user pressing key on element identified by `locator`.
 
-        `key` is either a single character, or a numerical ASCII code of the key
-        lead by '\\\\'.
+        `key` is either a single character, a numerical ASCII code of the key,\
+         or a NAMED KEY as described at https://selenium.googlecode.com/git/docs/api/py/webdriver/selenium.webdriver.common.keys.html
 
         Examples:
-        | Press Key | text_field   | q |
-        | Press Key | login_button | \\\\13 | # ASCII code for enter key |
-        """
-        if key.startswith('\\') and len(key) > 1:
-            key = self._map_ascii_key_code_to_key(int(key[1:]))
-        #if len(key) > 1:
-        #    raise ValueError("Key value '%s' is invalid.", key)
+        | Press Key | text_field   | q        |                                               |
+        | Press Key | login_button | \\\\13   | # ASCII code for Enter key (DEPRECATED)       |
+        | Press Key | login_button | 10       | # ASCII code for Return key                   |
+        | Press Key | nav_console  | ARROW_UP | # selenium.webdriver.common.keys ARROW_UP KEY |
+        
+        NAMED KEY value is new in Selenium2Library 1.7.3.
+        """
+        if len(key) > 1:
```

Comment:
I think that the sending numeric values below 10 should be only possible
with their NAMED values. But keeping the \\ would be OK.

On Wed, Jul 29, 2015 at 4:27 PM, Zephraph <notifications@github.com> wrote:

> In src/Selenium2Library/keywords/_element.py
> <https://github.com/rtomac/robotframework-selenium2library/pull/474#discussion_r35772772>
> :
>
> >
> >          Examples:
> > -        | Press Key | text_field   | q |
> > -        | Press Key | login_button | \\\\13 | # ASCII code for enter key |
> > -        """
> > -        if key.startswith('\\') and len(key) > 1:
> > -            key = self._map_ascii_key_code_to_key(int(key[1:]))
> > -        #if len(key) > 1:
> > -        #    raise ValueError("Key value '%s' is invalid.", key)
> > +        | Press Key | text_field   | q        |                                               |
> > +        | Press Key | login_button | \\\\13   | # ASCII code for Enter key (DEPRECATED)       |
> > +        | Press Key | login_button | 10       | # ASCII code for Return key                   |
> > +        | Press Key | nav_console  | ARROW_UP | # selenium.webdriver.common.keys ARROW_UP KEY |
> > +
> > +        NAMED KEY value is new in Selenium2Library 1.7.3.
> > +        """
> > +        if len(key) > 1:
>
> The one thing we have to consider with this approach is that users will no
> longer be able to specify 0-9 ASCII codes. The only ones that I think would
> be remotely important in that range are 8 and 9 (backspace and tab,
> respectively).
>
> Is this an issue? They could simply use BACKSPACE or TAB to do the same
> thing, so I'm not necessarily of the opinion that it really matters.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rtomac/robotframework-selenium2library/pull/474/files#r35772772>
> .
>